frontend/app/components/menu.tsx

```diff
@@ -1,25 +1,19 @@
 import type { ComponentProps } from 'react';
-import { useState } from 'react';
 
 import { useTranslation } from 'react-i18next';
 
+import { AppLink } from '~/components/app-link';
 import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from '~/components/dropdown-menu';
-import { InlineLink } from '~/components/inline-link';
 import { cn } from '~/utils/tailwind-utils';
 
-type MenuItemProps = ComponentProps<typeof InlineLink>;
+type MenuItemProps = ComponentProps<typeof AppLink>;
 
 export function MenuItem({ children, ...props }: MenuItemProps) {
   return (
-    <DropdownMenuItem className="text-md m-0 p-0 hover:!bg-slate-700">
-      <InlineLink
-        role="menuitem"
-        id="menu-item"
-        className="w-full px-3 py-2 text-white hover:bg-slate-300 hover:text-blue-950 focus:bg-slate-600 focus:text-white active:bg-slate-800 active:text-white"
-        {...props}
-      >
+    <DropdownMenuItem asChild className="cursor-pointer">
+      <AppLink data-testid="menu-item" {...props}>
```

Comment:
I'm not a fan of using `data-testid` in components like this. Instead, my preferred approach is to add the attribute in the test. Something like this:

```
const RoutesStub = createRoutesStub([
  {
    path: '/',
    Component: () => (
      <DropdownMenuItem data-testid="foo">
        ...
      </DropdownMenuItem>
    ),
  },
]);
```

Then run your assertions using the provided testid.

Reply:
Your approach will work since you can assign an attribute to it. However, when dealing with a component that has children composition and you need to perform assertions on its children, it's better to use data-testid.

Reply:
Use the testid to target the component being tested, then isolate any children as necessary using selectors.

Why unnecessarily clutter the production html if we don't have to?

Reply:
`data-testid` can be usefull when you're writing e2e tests. Another team could be responible to write those tests and could struggle to select elements. We had this issue with other projects before.